pkg/transformers/transformer.go

```diff
@@ -10,3 +10,28 @@ type Transformer interface {
 	// Transform Magistrala message to any other format.
 	Transform(msg *messaging.Message) (interface{}, error)
 }
+
+type number interface {
+	uint64 | int64 | float64
+}
+
+// ToUnixNano converts time to UnixNano time format.
+func ToUnixNano[N number](t N) N {
+	if t == 0 {
+		return 0
+	}
+	// Check if the value is in nanoseconds
+	if t > 1e18 {
```

Comment:
Should this be `>=`? We need either `>=` or `<=` in next cases to cover _equal_ scenario.

Reply:
By the way, it's perfectly fine to use literals here over constants. If nothing, it makes it more readable since you access values directly,